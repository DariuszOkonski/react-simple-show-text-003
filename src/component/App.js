import React, { Component } from 'react';
import './App.css';

const loremText = 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo? Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo?'

class App extends Component {
  state = {
    isTextShown: true
  }

  handleClick = () => {
    this.setState((prevState) => ({
      isTextShown: !prevState.isTextShown,
    }))
  }

  render() {
    return (
      <div className="app">
        <h1>Show text app</h1>
        <button onClick={this.handleClick}>{this.state.isTextShown ? 'Hide' : 'Show'}</button>
        {this.state.isTextShown && <p>{loremText}</p>}
      </div>
    );
  }
}

export default App;
