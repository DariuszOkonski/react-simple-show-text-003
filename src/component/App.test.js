import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App tests', () => {

  it('App renders without a problem', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Snapshot matches', () => {
    const wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot();
  });

  it('h1 exists', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find('h1').exists()).toBe(true);
    expect(wrapper.find('h1').text()).toBe('Show text app');
  });

  it('button exists', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find('button').exists()).toBe(true);
  });

  it('button changes text according to state', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find('button').text()).toBe('Hide');
    wrapper.setState({
      isTextShown: false
    })
    expect(wrapper.find('button').text()).toBe('Show');
  });

  it('handleClick', () => {
    const wrapper = shallow(<App />);

    expect(wrapper.find('button').text()).toBe('Hide');
    expect(wrapper.find('p').exists()).toBe(true);

    wrapper.find('button').simulate('click');

    expect(wrapper.find('button').text()).toBe('Show');
    expect(wrapper.find('p').exists()).toBe(false);
  });

  it('p exists', () => {
    const wrapper = shallow(<App />);

    wrapper.setState({
      isTextShown: true
    })

    expect(wrapper.find('p').exists()).toBe(true);

    wrapper.setState({
      isTextShown: false
    })

    expect(wrapper.find('p').exists()).toBe(false);
  });

});