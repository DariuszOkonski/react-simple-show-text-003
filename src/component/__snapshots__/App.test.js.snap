// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Snapshot matches 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Show text app
        </h1>,
        <button
          onClick={[Function]}
        >
          Hide
        </button>,
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo? Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo?
        </p>,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Show text app",
        },
        "ref": null,
        "rendered": "Show text app",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hide",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Hide",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo? Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo?",
        },
        "ref": null,
        "rendered": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo? Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo?",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Show text app
          </h1>,
          <button
            onClick={[Function]}
          >
            Hide
          </button>,
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo? Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo?
          </p>,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show text app",
          },
          "ref": null,
          "rendered": "Show text app",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hide",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Hide",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo? Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo?",
          },
          "ref": null,
          "rendered": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo? Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis assumenda similique sit qui suscipit. Recusandae ipsa quae impedit molestiae pariatur nam autem saepe voluptate fugiat? Tempora cumque fuga ab illo?",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
